File: src/08-challenge/solution.py
from typing import TypedDict

CharCounter = TypedDict("CharCounter", {"char": str, "amount": int})


def fn(toy: str) -> str:
    char_counters: list[CharCounter] = []

    for char in toy:
        index = -1
        for [i, char_counter] in enumerate(char_counters):
            if char.lower() == char_counter["char"].lower():
                index = i

        if index == -1:
            char_counters.append({"char": char, "amount": 1})
        else:
            char_counters[index]["amount"] += 1

    for char_counter in char_counters:
        if char_counter["amount"] < 2:
            return char_counter["char"]

    return ""

File: src/08-challenge/solution.ts
type CharCounter = {
	char: string
	amount: number
}

export default function fn(toy: string): string {
	const charCounters: CharCounter[] = []

	for (const char of toy) {
		const index = charCounters.findIndex((counter) => counter.char.toLowerCase() === char.toLowerCase())
		index === -1 ? charCounters.push({ char, amount: 1 }) : charCounters[index].amount++
	}

	const firstNonRepeatedChar = charCounters.find((counter) => counter.amount < 2)

	return firstNonRepeatedChar ? firstNonRepeatedChar.char : ''
}
